jina/types/arrays/document.py

```diff
@@ -299,6 +300,77 @@ def overriden_key(proto):
         else:
             self._pb_body.sort(*args, **kwargs)
 
+    def _compute_dist(
+        self, X: 'np.ndarray', Y: 'np.ndarray', metric: str
+    ) -> 'np.ndarray':
+        """Computes the distance between each row of X and each row on Y.
+
+        - Let `n_X = X.shape[0]`
+        - Let `n_Y = Y.shape[0]`
+        - Returns a matrix `dist` of shape `(n_X, n_Y)`.
+          `dist[i,j] = metric(X[i], X[j])`.
+
+        :param X: np.ndarray of ndim 2
+        :param Y:  np.ndarray of ndim 2
+        :param metric: string describing the metric type
+        :return:  np.ndarray of ndim 2
+        """
+        if metric == 'cosine':
+            dists = cosine_distance(X, Y)
+        if metric == 'euclidean':
+            dists = euclidean_distance_squared(X, Y)
+        return dists
+
+    def match(
+        self,
+        darray: 'DocumentArray',
+        metric: str = 'cosine',
+        limit: Optional[int] = None,
+    ) -> None:
+        """Compute embedding based nearest neighbour in `another` for each Document in `self`,
+        and store results in `matches`.
+
+        Note:
+            - If metric is 'cosine' it uses the cosine **similarity**.
+            - IF metric is 'euclidean' it uses the euclidean **distance**.
+
+        :param darray: the other DocumentArray to match against
+        :param metric: the distance metric
+        :param limit: the maximum number of matches, when not given
+                      all Documents in `another` are considered as matches
+        """
+        X = np.stack(self.get_attributes('embedding'))
+        Y = np.stack(darray.get_attributes('embedding'))
+        if limit is None:
```

Comment:
```suggestion
limit = limit or len(darray)
```

Reply:
updated in a single line with ` limit = min(limit, len(darray))
`